### DISCUSSION TITLE
有一个粉丝牌无法完成直播任务

### DISCUSSION THREAD

**Discussion by kevinkejc** [andywang425/BLTH#445] [General]
### 描述bug

具体的话是这两天都有一个粉丝牌无法完成直播任务，并且这两天无法完成任务的牌子是不一样的，也检查过直播间是有选分区没问题的

这两个牌子在这两天的共同点是在装扮中心里都位于第二页第一个
![image](https://github.com/andywang425/BLTH/assets/75735934/73cb1028-fc0a-4f8a-8ce8-8d7eb3136617)
点赞任务和发弹幕任务是我把脚本存储里对应任务的`_lastCompleteTime`数字改掉之后才完成的(但是观看直播任务用这个方法就不行了，短暂运行了一下之后就显示观看直播任务完成)

以及不知道是B站抽风了还是说我出问题了，我明明有19个牌子，但是在装扮中心里只看到了18个(牌子并没有被删除，去消失牌子的直播间里看牌子还是在的)，在脚本的黑白名单里也无法查看到在装扮中心里消失的那个牌子，而且消失的这个牌子刚好是昨天无法完成直播任务的那个牌子(今天这个消失的牌子完成了直播任务)
![image](https://github.com/andywang425/BLTH/assets/75735934/e3f49f86-1a40-4b53-be89-ff328170fa26)
![image](https://github.com/andywang425/BLTH/assets/75735934/eaffae70-8566-47ba-8cef-749bc8711417)
![image](https://github.com/andywang425/BLTH/assets/75735934/33eb5b45-7a2f-4b34-a88c-2d6c612703c3)

### 重现bug

脚本执行完任务之后查看粉丝牌亲密值的时候发现的，没进行什么操作

### 预期行为

粉丝牌满亲密值

### 运行日志

![image](https://github.com/andywang425/BLTH/assets/75735934/debae08c-1b77-45fb-ab44-749900ad5e3f)

### 操作系统

Windows 10 专业版 21H2

### 浏览器

Google Chrome 115.0.5790.171（正式版本）（64 位）

### 用户脚本管理器

Tampermonkey v4.19.0

### 脚本版本号

7.0.6

### 网络情况

好

### 其他

给的日志并不是今天首次执行任务的日志，是把`_lastCompleteTime`改过之后的新日志
![image](https://github.com/andywang425/BLTH/assets/75735934/10bfc892-2328-49cc-8ef8-8b92131ea9f5)
就改的这三个任务

在控制台里直接就搜索不到那个消失掉的粉丝牌的执行日志了
![image](https://github.com/andywang425/BLTH/assets/75735934/34ba7783-a5fd-4bbb-b4c2-d898d7a80a5b)

那个无法完成任务的粉丝牌还是可以搜索得到的
![image](https://github.com/andywang425/BLTH/assets/75735934/f3f4e548-53d9-424e-b551-8dccc5354879)

也不知道这个消失掉的粉丝牌任务是怎么完成的.....

**Comment by andywang425** [comment]
我觉得需要进一步分析获取粉丝勋章API的response。你在粉丝勋章相关任务还没完成时打开控制台，搜索`BAPI.live.fansMedalPanel`应该能找到几条日志，按顺序鼠标右键点击箭头右边的那些数据->点击“复制object"，然后黏贴发出来。你有两页粉丝勋章的话应该就是要复制黏贴两次。这其中包含你的UID信息，如果你介意的话可以发到我邮箱 andywang425@qq.com 。
![image](https://github.com/andywang425/BLTH/assets/56548456/335eeaf2-3bed-4deb-8e5a-53a4af837cd6)

顺便一提，对于观看直播任务来说光是改`_lastCompleteTime`是不够的，这个任务的执行还和`_lastWatchTime`以及`_watchedSecondsToday`有关。如果想要再执行一次观看直播任务，你可以把它们全改成0。
![image](https://github.com/andywang425/BLTH/assets/56548456/4e830acd-6100-4f3a-9f82-8efd40ff77db)

**Comment by kevinkejc** [comment]
已经把两条日志发到邮箱了

我用了新临时解决方法之后就可以正常增长亲密值了

**Comment by andywang425** [comment]
邮件已收到。你这个情况确实奇怪，API的返回结果里写着一共有19个粉丝勋章，但确实只有18个。而且我看了下第二页第一个勋章的相关数据也没什么问题。根据你之前提供的日志，这个粉丝勋章（房间号`482857`）在执行任务时也没漏掉。我建议你试试看能不能通过正常完成任务的方式给这个勋章加亲密度。

你那个消失的粉丝勋章，它的数据不在API的返回结果里，只能是B站的问题了。

综上我认为这不算是脚本的Bug。粉丝勋章消失这个问题可能得找客服。

**Comment by kevinkejc** [comment]
看着是B站的问题了，我重新执行完一次直播任务之后那个粉丝勋章就正常加亲密值了，....然后我发现那个在之前无法完成任务的粉丝勋章(房间号482857)在装扮中心里消失了，消失的那个粉丝牌反而回来了，然后现在装扮中心里还是只有18个牌子，属实是B站的疑难杂症了，还是得多观察几天看看

**Comment by kevinkejc** [comment]
经过几天的观察，大概率是我这满级牌和舰长牌的原因，之前满级牌和舰长牌点亮的期间在装扮中心就只能看到18个牌子，这就导致漏掉一个牌子的亲密值没加，得重置一次直播任务才行，现在满级牌和舰长牌熄灭了，就可以在装扮中心里看到所有牌子了，也就可以给全部小于20级的粉丝牌加满亲密值了

![image](https://github.com/andywang425/BLTH/assets/75735934/1f4cf633-2bb9-4c92-ba69-7d23eb203f1f)
顺便反馈下开了这个功能好像是无法点亮粉丝勋章的，点赞直播间功能完全正常运行，也能看到加的100亲密值
![image](https://github.com/andywang425/BLTH/assets/75735934/dde3374c-ac39-49b4-99d2-c20a74a612dc)
 但就是很奇怪的不能点亮粉丝勋章，属实是被帮助中心给坑了

**Comment by andywang425** [comment]
B站的帮助中心更新不及时，现在确实是没法通过点赞点亮勋章了。

**Comment by kevinkejc** [comment]
在今天，满级牌和舰长牌熄灭的情况下又只能在装扮中心看到18个粉丝勋章，然后也是跟之前一样漏掉了一个牌子亲密值完全没加，也只能再执行一次直播任务，看起来不是≥20级粉丝勋章的问题了，这些天再找些牌子卡一下试试了

**Comment by kevinkejc** [comment]
测试了下，卡新粉丝勋章也是没用处的，也就只能把消失的粉丝勋章给戴起来才恢复正常，但这样就导致了没法在某些有开粉丝勋章禁言的直播间发弹幕了，总体来讲问题不大，多看五分钟直播的事，就是不知道一直不发弹幕会不会熄灭

**Comment by andywang425** [comment]
既然不是BLTH的bug，我就把issue转成discussion了。如果还有后续的话在这里更新吧。
